pylanetary/navigation/core.py

```diff
@@ -826,56 +826,45 @@ def reproject(self, pixscale_arcsec = None, interp = 'cubic'):
-        print('New image will be %d by %d pixels'%(2*npix + 1, npix))
-        print('Pixel scale %f km = %f pixels per degree'%(self.pixscale_km * (pixscale_arcsec / self.pixscale_arcsec), npix_per_degree))
-        
-        #create new lon-lat grid
-        extra_wrap_dist = 180
-        newlon, newlat = np.arange(-extra_wrap_dist,360 + extra_wrap_dist, 1/npix_per_degree), np.arange(-90,90, 1/npix_per_degree)
-        gridlon, gridlat = np.meshgrid(newlon, newlat)
-        nans = np.isnan(self.lon_w.flatten())
-        def input_helper(arr, nans):
-            '''removing large region of NaNs speeds things up significantly'''
-            return arr.flatten()[np.logical_not(nans)]
-        inlon, inlat, indat = input_helper(self.lon_w, nans), input_helper(self.lat_g, nans), input_helper(self.data, nans)
-
-        #fix wrapping by adding dummy copies of small lons at > 360 lon
-        inlon_near0 = inlon[inlon < extra_wrap_dist]
-        inlon_near0 += 360
-        inlon_near360 = inlon[inlon > 360 - extra_wrap_dist]
-        inlon_near360 -= 360
-        inlon_n = np.concatenate((inlon_near360, inlon, inlon_near0))
-        inlat_n = np.concatenate((inlat[inlon > 360 - extra_wrap_dist], inlat, inlat[inlon < extra_wrap_dist]))
-        indat_n = np.concatenate((indat[inlon > 360 - extra_wrap_dist], indat, indat[inlon < extra_wrap_dist]))
-
-        #do the regridding
-        datsort = griddata((inlon_n, inlat_n), indat_n, (gridlon, gridlat), method = interp)
-        
-        #trim extra data we got from wrapping
-        wrap_i_l = len(gridlon[0][gridlon[0] < 0]) - 1
-        wrap_i_u = len(gridlon[0][gridlon[0] >= 360])
-        datsort = datsort[:,wrap_i_l:-wrap_i_u]
-        gridlon = gridlon[:,wrap_i_l:-wrap_i_u]
-        gridlat = gridlat[:,wrap_i_l:-wrap_i_u]
-        
-        # make far side of planet into NaNs
-        snorm = surface_normal(gridlat, gridlon, self.ob_lon)
-        #emang = emission_angle(self.ob_lat, snorm).T
-        emang = emission_angle(self.ob_lat, snorm)
-        farside = np.where(emang < 0.0)
-        datsort[farside] = np.nan
-        emang[farside] = np.nan
-        projected = datsort
-        mu_projected = emang
-        
-        self.projected_data = projected
-        self.projected_mu = mu_projected
-
-        return projected, mu_projected
+        if shape is not None:
+            target_nx = int(shape[0])
+            target_ny = int(shape[1])
+        if type(projection) == str:
+            projection = projection.lower()
+        
+        # compute source coordinates in kilometer and specify projection type
+        img_globe = ccrs.Globe(semimajor_axis=self.req , semiminor_axis=self.rpol, ellipse=None)
+        source_proj = ccrs.Orthographic(central_longitude=self.ob_lon, central_latitude=self.ob_lat, globe=img_globe) 
+        input_nx, input_ny = self.data.shape[0], self.data.shape[1]
+            
+        if projection == 'equirectangular': 
```

Comment:
Maybe a bit more comments here for what options are doing what exactly

Reply:
It seems clear from the code that this is applying an equirectangular projection. To better explain what that does, I added some additional stuff in the docstring. I have been trying to follow the suggestions of [Code Aesthetic](https://www.youtube.com/watch?v=Bf7vDBBOBUA&ab_channel=CodeAesthetic) about comments: normally if a comment is required inside the code, then the code wasn't written clearly enough. The docstring is where it should be explained to a non-developer what is happening.  Let me know if you want more changes here.